### ISSUE TITLE
Remove spaces from correct answer when remove-spaces is "true"

### ISSUE THREAD

**Issue by bowmnath** [PrairieLearn/PrairieLearn#2165]
The remove-spaces attribute of pl-string-input currently
- removes spaces from the student answer
- compares stripped student answer against correct-answer

I think it would be better to
- remove spaces from the student answer _and_ the correct-answer
- compare stripped student answer against stripped correct-answer

Otherwise, a correct-answer with spaces is impossible for a student to match.  However, it might be nice to have those spaces in the correct-answer for displaying.  Also, this keeps someone writing a question from shooting themself in the foot by adding spaces to correct-answer.

It is difficult to imagine a case in which this change would be harmful.  The only way it would break backward compatability is if someone put spaces in a supposedly space-free answer in order to deliberately make the question impossible, which doesn't seem likely.

**Comment by coatless** [comment]
@nicknytko maybe tackle after #2065 goes in?

**Comment by nicknytko** [comment]
I'd think we would also want a similar check for the `remove-leading-trailing` attribute, yes?
